optuna_dashboard/ts/components/GraphRank.tsx

```diff
@@ -0,0 +1,342 @@
+import * as plotly from "plotly.js-dist-min"
+import React, { FC, useEffect, useState } from "react"
+import {
+  Grid,
+  FormControl,
+  FormLabel,
+  MenuItem,
+  Select,
+  Typography,
+  SelectChangeEvent,
+  useTheme,
+  Box,
+} from "@mui/material"
+import { plotlyDarkTemplate } from "./PlotlyDarkMode"
+import { makeHovertext } from "../graphUtil"
+import { useMergedUnionSearchSpace } from "../searchSpace"
+
+const PADDING_RATIO = 0.05
+const plotDomId = "graph-rank"
+
+interface AxisInfo {
+  name: string
+  range: [number, number]
+  is_log: boolean
+  is_cat: boolean
+}
+
+interface RankPlotInfo {
+  xaxis: AxisInfo
+  yaxis: AxisInfo
+  xvalues: (string | number)[]
+  yvalues: (string | number)[]
+  zvalues: number[]
+  colors: number[]
+  hovertext: string[]
+}
+
+export const GraphRank: FC<{
+  study: StudyDetail | null
+}> = ({ study = null }) => {
+  const theme = useTheme()
+  const [objectiveId, setobjectiveId] = useState<number>(0)
+  const searchSpace = useMergedUnionSearchSpace(study?.union_search_space)
+  const [xParam, setXParam] = useState<SearchSpaceItem | null>(null)
+  const [yParam, setYParam] = useState<SearchSpaceItem | null>(null)
+  const objectiveNames: string[] = study?.objective_names || []
+
+  if (xParam == null && searchSpace.length > 0) {
+    setXParam(searchSpace[0])
+  }
+  if (yParam == null && searchSpace.length > 1) {
+    setYParam(searchSpace[1])
+  }
+
+  const handleObjectiveChange = (event: SelectChangeEvent<number>) => {
+    setobjectiveId(Number(event.target.value))
+  }
+  const handleXParamChange = (event: SelectChangeEvent<string>) => {
+    const param = searchSpace.find((item) => item.name === event.target.value)
+    setXParam(param || null)
+  }
+  const handleYParamChange = (event: SelectChangeEvent<string>) => {
+    const param = searchSpace.find((item) => item.name === event.target.value)
+    setYParam(param || null)
+  }
+
+  const rankPlotInfo = getRankPlotInfo(study, objectiveId, xParam, yParam)
+
+  useEffect(() => {
+    if (study != null) {
+      plotRank(rankPlotInfo, theme.palette.mode)
+    }
+  }, [study, theme.palette.mode])
```

Comment:
To change ui in a timely manner, we need to change those lines something like this. Otherwise, when we change objectiveId, xParam and yParam, the ui will not be changed immediately.

```
  useEffect(() => {
    if (study != null) {
      const rankPlotInfo = getRankPlotInfo(study, objectiveId, xParam, yParam)
      plotRank(rankPlotInfo, theme.palette.mode)
    }
  }, [study, objectiveId, xParam, yParam, theme.palette.mode]
```